### DISCUSSION TITLE
Union of legit types is also a legit type

### DISCUSSION THREAD

**Discussion by sassanh** [python/typing#1796] [Q&A] [answered]
Is there anyway in Python's typing system to have something like this work:

```python
class OriginalType: ...
class SubclassOne(OriginalType): ...
class SubclassTwo(OriginalType): ...
UnionType = SubclassOne | SubclassTwo

def some_function(_: type[OriginalType]): ...

some_function(UnionType)
```

I want to be able to limit the argument of my function to a specific union of arbitrary subclasses of the base class. A real life usage of such function is here: https://github.com/sassanh/python-redux/blob/main/tests/test_features.py#L127

Context: https://github.com/microsoft/pyright/issues/8390

Is it not possible until https://peps.python.org/pep-0747/ is done?

**Comment by JelleZijlstra** [comment, marked as answer]
Yes, this is a use case for PEP-747. A union is not an instance of `type`, so it is not assignable to `type[]`.
